### DISCUSSION TITLE
Custom js does not run in action modal - Uncaught ReferenceError: function is not defined

### DISCUSSION THREAD

**Discussion by joshembling** [filamentphp/filament#9393] [General]
I have a really simple view field which I am rendering within a custom action. The HTML is rendered from my test.blade.php file, but anything within the script tags is not recognised and I am getting the "Uncaught ReferenceError: function is not defined" message in the console.

```php
Tables\Actions\Action::make('myAction')
->requiresConfirmation()
->form([
    ViewField::make('name')
        ->view('test'),
])
```

```html
// test.blade.php
<script>
console.log('here') // this is ignored

function test() { 
    console.log('hello world') // never gets called
}
</script>

<button onclick="test">Click!</button> // renders fine
```

`Uncaught ReferenceError: test is not defined at HTMLButtonElement.onclick`

This doesn't seem to be a problem when rendering the same thing within a regular form outside of a modal. It seems that anything within script tags does not execute whenever in a modal.

Does anyone know why this is the case and how I can fix?

**Comment by leandrocfe** [comment]
Why not using alpinejs?
https://alpinejs.dev/start-here#building-a-counter

**Reply by joshembling** [reply]
My real use case is a lot more complex than my example. However, I can't even get the example working inside a modal, so my real use case (using third party libraries, firing to third party sources, pusher integration) is all written in vanilla js, and works outside of a modal. 

It seems wrong that I should have to rewrite everything to use alpine instead.

**Comment by joshembling** [comment]
Anyone else with any suggestions?

**Comment by Kekuatan** [comment]
i have same problem to
Some suggest... 

js file
 ``` javascript
 function convertNumber(classSelector, countryString,isKurs=false, withoutDecimal=false){
        for (let i = 0, len = classSelector.length; i < len; i++) {
            let str = classSelector[i].innerHTML
            let digit = withoutDecimal ? 0 : 2;
            let num = Number(str).toLocaleString(countryString, {minimumFractionDigits: digit, maximumFractionDigits: digit})


            //req ci nonih replace jika .00-
            if (isKurs)
            {
                classSelector[i].innerHTML = num;
            }
            else
            {
                classSelector[i].innerHTML = num.replace(",00", "");
            }
        }
    }
```

Filament file
``` php
Action::make($actionName)
      ->mountUsing(function ($livewire) {
               $livewire->js('
                       let x = document.querySelectorAll(".number");
                       convertNumber(x, "id-ID");
                ');
            })
```
no luck for me by using that

but i solve it by using alpine.js to call my function

``` html
    <div x-data="{ formatNumber: function(number, countryString = 'id-ID', isKurs = false, withoutDecimal = false) {
        let digit = withoutDecimal ? 0 : 2;
        let num = Number(number).toLocaleString(countryString, { minimumFractionDigits: digit, maximumFractionDigits: digit });
        if (isKurs) {
            return num;
        } else {
            return num.replace(',00', '');
        }
    }}" x-init="() => {
        const numbers = document.querySelectorAll('.number');
        numbers.forEach(element => {
            let number = element.innerHTML;
            element.innerHTML = formatNumber(number);
        });
    }">
    </div>
  ```
  and put that hollow div somewhere on my blade

Maybe it can help...

**Reply by djped** [reply]
Just battling with this myself, thanks for the tip. Works great!

**Comment by joseayramdinardi** [comment]
@djped I know that you solved your problem, but, I think that the right way is https://filamentphp.com/docs/3.x/support/assets#registering-javascript-files
